evm_arithmetization/src/cpu/kernel/asm/main.asm

```diff
@@ -85,15 +85,18 @@ global execute_withdrawals:
 execute_withdrawals_post_stack_op:
     %withdrawals
 
-global hash_final_tries:
+global perform_final_checks:
     // stack: cum_gas, txn_counter, num_nibbles, txn_nb
     // Check that we end up with the correct `cum_gas`, `txn_nb` and bloom filter.
     %mload_global_metadata(@GLOBAL_METADATA_BLOCK_GAS_USED_AFTER) %assert_eq
     DUP3 %mload_global_metadata(@GLOBAL_METADATA_TXN_NUMBER_AFTER) %assert_eq
     %pop3
-    PUSH 1 // initial trie data length 
+    PUSH 1 // initial trie data length
+global check_state_trie:
     %mpt_hash_state_trie   %mload_global_metadata(@GLOBAL_METADATA_STATE_TRIE_DIGEST_AFTER)     %assert_eq
+global check_txn_trie:
     %mpt_hash_txn_trie     %mload_global_metadata(@GLOBAL_METADATA_TXN_TRIE_DIGEST_AFTER)       %assert_eq
+global check_receipt_trie:
```

Comment:
Yeah I guess we have been adding these in locally for so long that maybe it makes more sense just to get this into main. :upside_down_face: